RecoEgamma/PhotonIdentification/plugins/PhotonMVAEstimator.cc

```diff
@@ -61,22 +61,29 @@ PhotonMVAEstimator::
 }
 
 float PhotonMVAEstimator::
-mvaValue(const edm::Ptr<reco::Candidate>& particle, const edm::EventBase& iEvent) const {
+mvaValue(const edm::Ptr<reco::Candidate>& candPtr, const edm::EventBase& iEvent) const {
+
+  const int iCategory = findCategory( candPtr );
```

Comment:
You should also check that the `iCategory` is not -1 since, if it is, you will be reading off the end of the vector.

Reply:
Actually, it looks like `findCategory` returne `UNDEFINED` if it fails.

Reply:
And UNDEFINED is exactly -1:
https://github.com/guitargeek/cmssw/blob/1b25fbf6ac721c8a69d9ce94cd82479e8e5e252f/RecoEgamma/PhotonIdentification/interface/PhotonMVAEstimator.h#L30

But it can only return -1 if eta is in the border and in the unlikely case of rounding problems... Still a fix can be introduced for it